flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -1601,10 +1601,15 @@ private OptionalLong restoreLatestCheckpointedStateInternal(
         // Create the new operator states without in-flight data.
         for (OperatorState originalOperatorState : originalOperatorStates.values()) {
             OperatorState newState =
-                    new OperatorState(
-                            originalOperatorState.getOperatorID(),
-                            originalOperatorState.getParallelism(),
-                            originalOperatorState.getMaxParallelism());
+                    originalOperatorState.isFullyFinished()
+                            ? new FullyFinishedOperatorState(
+                                    originalOperatorState.getOperatorID(),
+                                    originalOperatorState.getParallelism(),
+                                    originalOperatorState.getMaxParallelism())
+                            : new OperatorState(
+                                    originalOperatorState.getOperatorID(),
+                                    originalOperatorState.getParallelism(),
+                                    originalOperatorState.getMaxParallelism());
```

Comment:
I would extract this logic (and some lines below) to a method like `OperatorState OperatorState#discardInFlightData()` method, that would return the `newState`. It would be less error prone  to have this logic encapsulated in the `OperatorState` class.

Reply:
Very thanks for the suggestions! It makes a lot of senses to me~ I modified the code accordingly~